dev-tools/gen-target-info/src/main.rs

```diff
@@ -0,0 +1,29 @@
+use gen_target_info::{get_target_specs_from_json, write_string_mapping_to_file};
+use std::{fs::File, io::Write as _};
+
+fn main() {
+    let target_specs = get_target_specs_from_json();
+
+    // Open file to write to
+    let manifest_dir = env!("CARGO_MANIFEST_DIR");
+
+    let path = format!("{manifest_dir}/../../src/target_info.rs");
+    let mut f = File::create(path).expect("failed to create src/target_info.rs");
+
```

Comment:
Pursuant to my comment elsewhere, we can write the "header comment" here before we do any more writes.

Reply:
Thanks, wrote the header as suggested.